### PR TITLE
fix: honor use_aws_shared_config_files

### PR DESCRIPTION
**Pull request by yenfryherrerafeliz**
Honor the `use_aws_shared_config_files` config from `default_region` and `default_sigv4a_signing_region_set` by passing `$args` along to the `ConfigurationResolver::resolve` call.

*Issue #3026*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tests/ClientResolverTest.php
Comment (stobrien89) [inline tests/ClientResolverTest.php:1]: These tests can be removed— we already have unit tests for this in `ConfigurationResolverTest`

**Review by stobrien89 — CHANGES_REQUESTED** [review]
Just a request to remove the tests— we cover that behavior in `ConfigurationResolverTest` and this test class is arguably bloated already.

File: tests/ClientResolverTest.php
Comment (stobrien89) [reply tests/ClientResolverTest.php:1]: Per our discussion, I agree that moving `resolve` calls to a method on the `ClientResolver` would help enforce passing `$args`.

File: src/ClientResolver.php [hunk 1]
```diff
@@ -1356,6 +1345,36 @@ public static function _missing_region(array $args)
...
+     *
+     * @param $key
+     * @param $expectedType
+     * @param $args
+     *
+     * @return mixed|string
+     */
+    private static function _resolve_from_env_ini(
+        $key,
+        $expectedType,
```
Comment (stobrien89) [inline src/ClientResolver.php:1359]: Let's add some type hints here

File: src/ClientResolver.php [hunk 2]
```diff
@@ -327,7 +341,7 @@ class ClientResolver
             'valid' => ['array', 'string'],
```
Comment (stobrien89) [inline src/ClientResolver.php:341]: The configuration resolver technically doesn't handle arrays (nor should it), so I think we'll need to move 'string' to the front here.

File: src/ClientResolver.php [hunk 3]
```diff
@@ -1356,6 +1345,36 @@ public static function _missing_region(array $args)
...
+        $key,
+        $expectedType,
+        $args
+    ) {
+        static $typeDefaultMap = [
+            'int' => 0,
+            'bool' => false,
+            'boolean' => false,
+            'string' => '',
+            'array' => []
```
Comment (stobrien89) [inline src/ClientResolver.php:1367]: Let's remove 'array'

**Review by stobrien89 — CHANGES_REQUESTED** [review]

File: src/ClientResolver.php (hunk 2 above)
Comment (yenfryherrerafeliz) [reply src/ClientResolver.php:341]: Good catch. Thanks!

**Review by stobrien89 — APPROVED** [review]

**Pull request merged**